### PR TITLE
Dynamic entry type options fields

### PR DESCRIPTION
**Pull request by vassalloandrea**
Until now we can update the option_type adding the options using a custom JSON fields.
The problem is that we need to save different fields, which is mandatory, for each provider.

 With this PR, we added the ability to render the correct option form based on the chosen provider during the option type creation.
This means that the admin user can't change the provider of an option type if it has already been created.

### CODE REVIEW

File: app/models/solidus_content/entry_type.rb [hunk 1]
```diff
@@ -1,53 +1,63 @@
 # frozen_string_literal: true
 
-class SolidusContent::EntryType < ActiveRecord::Base
-  has_many :entries, dependent: :destroy
+module SolidusContent
```
Comment (elia) [inline app/models/solidus_content/entry_type.rb]: Not blocking, but would be better to keep the namespace change in a separate commit, so that the actual change is more clear

File: lib/solidus_content/providers/yaml.rb [hunk 2]
```diff
@@ -2,28 +2,28 @@
...
+          data = load(file)
 
-    input.merge(data: data)
-  end
+          input.merge(data: data)
+        end
 
-  # Ruby 2.6 deprecated passing the filename as the second argument, we'll do 
-  # a feature check instead of relying on the Ruby version.
-  filename_key = YAML.method(:load).parameters.rassoc(:filename).first == :key
```
Comment (elia) [inline lib/solidus_content/providers/yaml.rb:18]: Removing this we're losing contextualized stacktraces if there's a syntax error in the YAML, are we sure this version is preferable? Maybe it's worth mentioning this in the commit message as well.

File: lib/solidus_content/providers/renderful.rb [hunk 3]
```diff
@@ -1,17 +0,0 @@
...
-  end
-
-  def call(input)
-    entry_id = input.dig(:options, :id)
-
-    input.merge(data: {
-      render_in: -> (view_context) { renderful.render(entry_id, view_context: view_context) },
-    })
-  end
-end
```
Comment (elia) [inline lib/solidus_content/providers/renderful.rb:17]: I'd be interested in reading why it's useless in the commit description

**Review by elia — COMMENTED** [review]
A few comments, mostly nits and formal stuff

**Comment by elia** [comment]
BTW thanks for getting after and solving this problem! As already stated elsewhere this is really great stuff! 👏

File: lib/solidus_content/providers/renderful.rb (hunk 3 above)
Comment (vassalloandrea) [reply lib/solidus_content/providers/renderful.rb:17]: I'll update the commit message with more information

File: lib/solidus_content/providers/yaml.rb (hunk 2 above)
Comment (vassalloandrea) [reply lib/solidus_content/providers/yaml.rb:18]: To avoid the feature check on the `load` method, we can use `load_file` that isn't deprecated instead.
The problem is that `load_file` doesn't support `sym_keys` by default and I called it after the file loading.

**Review by elia — APPROVED** [review]
🎉
